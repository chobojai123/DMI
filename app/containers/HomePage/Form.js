import styled from 'styled-components';

const Form = styled.form`
  color: white;
  width: 60%;
  padding: 80px 0;
  margin: 0 auto;
`;

export default Form;
