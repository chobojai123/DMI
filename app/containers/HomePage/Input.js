import styled from 'styled-components';

const Input = styled.input`
  color: white;
  border: 1px solid white;
  border-radius: 3px;
  padding: 7px;
  width: 100%;
`;

export default Input;
