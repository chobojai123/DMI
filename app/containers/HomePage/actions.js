import {
  CREATE_POST,
  CREATE_POST_SUCCESS,
  CREATE_POST_ERROR,
} from './constants';

export function createPost(message) {
  return {
    type: CREATE_POST,
    message,
  };
}

export function createPostSuccess(message) {
  return {
    type: CREATE_POST_SUCCESS,
    message,
  };
}

export function createPostError(error) {
  return {
    type: CREATE_POST_ERROR,
    error,
  };
}
