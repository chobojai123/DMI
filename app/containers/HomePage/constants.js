export const CREATE_POST = 'boilerplate/HomePage/CREATE_POST';
export const CREATE_POST_SUCCESS = 'boilerplate/HomePage/CREATE_POST_SUCCESS';
export const CREATE_POST_ERROR = 'boilerplate/HomePage/CREATE_POST_ERROR';
