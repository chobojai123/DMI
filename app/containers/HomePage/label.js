import styled from 'styled-components';

const Input = styled.label`
  color: white;
  font-size: 120%;
  font-weight: 200;
`;

export default Input;
