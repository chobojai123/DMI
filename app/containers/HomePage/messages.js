import { defineMessages } from 'react-intl';

export default defineMessages({
  textAreaMessage: {
    id: 'boilerplate.containers.HomePage.textArea.Message',
    defaultMessage: 'WRITE A MESSAGE',
  },
});
