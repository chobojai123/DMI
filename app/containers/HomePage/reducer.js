import { fromJS } from 'immutable';

import {
  CREATE_POST,
  CREATE_POST_SUCCESS,
  CREATE_POST_ERROR,
} from './constants';

export const initialState = fromJS({
  message: false,
  messagePosted: false,
  postError: false,
});

function homeReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_POST:
      return state.set('message', action.message);
    case CREATE_POST_SUCCESS:
      return state.set('messagePosted', true);
    case CREATE_POST_ERROR:
      return state.set('postError', action.error);
    default:
      return state;
  }
}

export default homeReducer;
