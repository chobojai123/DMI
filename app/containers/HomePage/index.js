import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import { loadMessages, messagesLoaded } from 'containers/App/actions';
import { createPost } from './actions.js';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import reducer from './reducer';
import saga from './saga';

import messages from './messages';
import Form from './Form';
import Input from './Input';
import TextArea from './TextArea';
import Label from './Label';
import StyledButton from '../../components/Button/StyledButton';
import '../../css/grid.css';

class HomePage extends Component {
  constructor() {
    super();
    this.state = {
      message: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    this.props.fetchMessages();
  }

  handleChange(event) {
    this.setState({ message: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.onSubmitForm(this.state.message);
    this.setState({ message: '' });
  }

  render() {
    return (
      <div className="row">
        <Helmet>
          <title>Home Page</title>
          <meta
            name="description"
            content="A React.js Boilerplate application homepage"
          />
        </Helmet>
        <Form onSubmit={this.handleSubmit}>
          <div className="row">
            <div className="col span-3-of-3">
              <h2>
                <FormattedMessage {...messages.textAreaMessage} />
              </h2>
            </div>
          </div>
          <div className="row">
            <div className="col span-3-of-3">
              <TextArea
                type="text"
                name="firstname"
                placeholder="Your message"
                onChange={this.handleChange}
                value={this.state.message}
                required
              />
            </div>
          </div>
          <div className="row">
            <div className="col span-1-of-3">
              <StyledButton type="submit">Post it!</StyledButton>
            </div>
          </div>
        </Form>
      </div>
    );
  }
}

HomePage.propTypes = {
  onSubmitForm: PropTypes.func,
};

const mapDispatchToProps = dispatch => {
  return {
    onSubmitForm: message => dispatch(createPost(message)),
    fetchMessages: () => dispatch(loadMessages()),
  };
};

const withConnect = connect(
  null,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'home', reducer });
const withSaga = injectSaga({ key: 'home', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(HomePage);
