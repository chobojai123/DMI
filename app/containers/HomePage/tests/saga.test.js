import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_MESSAGES } from 'containers/App/constants';
import { CREATE_POST } from 'containers/HomePage/constants';
import { messagesLoaded, messagesLoadingError } from 'containers/App/actions';
import {
  createPostSuccess,
  createPostError,
} from 'containers/HomePage/actions';
import messageWatcherSaga, { getAllMessages, createMessage } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getAllMessages Saga', () => {
  let getMsgsGenerator;

  beforeEach(() => {
    getMsgsGenerator = getAllMessages();

    const callDescriptor = getMsgsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the messagesLoaded action if it requests the data successfully', () => {
    const response = '123';
    const putDescriptor = getMsgsGenerator.next(response).value.PUT.type;
    expect(putDescriptor).toEqual(put(messagesLoaded(response)).type);
  });

  it('should call the messagesLoadingError action if the response errors', () => {
    const response = new Error('error');
    const putDescriptor = getMsgsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(messagesLoadingError(response)));
  });
});

describe('createMessage Saga', () => {
  let messageGenerator;

  const messageToPost = { message: '123' };

  beforeEach(() => {
    messageGenerator = createMessage(messageToPost);

    const callDescriptor = messageGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the createPostSuccess action if it requests the data successfully', () => {
    const response = 'Successfully posted';
    const putDescriptor = messageGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(createPostSuccess(response)));
  });

  it('should call the createPostError action if the response errors', () => {
    const response = new Error('error');
    const putDescriptor = messageGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(createPostError(response)));
  });
});

describe('messageDataSaga Saga', () => {
  const messageDataSaga = messageWatcherSaga();

  it('should start task to watch for CREATE_POST and LOAD_MESSAGES action', () => {
    const takeLatestDescriptor = messageDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual([
      takeLatest(CREATE_POST, createMessage),
      takeLatest(LOAD_MESSAGES, getAllMessages),
    ]);
  });
});
