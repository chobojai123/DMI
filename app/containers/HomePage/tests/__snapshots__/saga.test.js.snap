// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMessage Saga should call the createPostError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`createMessage Saga should dispatch the createPostSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getAllMessages Saga should call the messagesLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getAllMessages Saga should dispatch the messagesLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;
